### PR TITLE
[SPARK-12646] support _HOST tag in kerberos principal

### PR DESCRIPTION
**Pull request by haridsv**

### CODE REVIEW

**Comment by haridsv** [comment]
Could someone review?

File: core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala [hunk 1]
```diff
@@ -126,7 +127,15 @@ class SparkHadoopUtil extends Logging {
...
   def loginUserFromKeytab(principalName: String, keytabFilename: String) {
-    UserGroupInformation.loginUserFromKeytab(principalName, keytabFilename)
+    var principal = SecurityUtil.getServerPrincipal(principalName, InetAddress.getLocalHost)
+    UserGroupInformation.loginUserFromKeytab(principal, keytabFilename)
+  }
+
+  def loginUserFromKeytabAndReturnUGI(
+        principalName: String,
+        keytabFilename: String): UserGroupInformation = {
+    var principal = SecurityUtil.getServerPrincipal(principalName, InetAddress.getLocalHost)
```
Comment (tgravescs) [inline core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala]: you can't do this for normal operation. A user principal is not the same as a server principal and normally users principals don't have HOSTS.  I don't see how this doesn't break the current functionality.

As I believe @vanzin stated in the jira _HOST is meant for servers, not for applications running on yarn.

**Comment by tgravescs** [comment]
Why don't you use one headless user keytab for all your apps?

what you are doing seems very weird for me.

**Comment by haridsv** [comment]
Thanks @tgravescs! Yes, I moved away from this approach to using delegation tokens. I updated the jira a few days back.

**Pull request closed without merging**